db/migrate/20181024224956_migrate_account_conversations.rb

```diff
@@ -0,0 +1,43 @@
+class MigrateAccountConversations < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def up
+    say ''
+    say 'WARNING: This migration may take a *long* time for large instances'
+    say 'It will *not* lock tables for any significant time, but it may run'
+    say 'for a very long time. We will pause for 10 seconds to allow you to'
+    say 'interrupt this migration if you are not ready.'
+    say ''
+
+    10.downto(1) do |i|
+      say "Continuing in #{i} second#{i == 1 ? '' : 's'}...", true
+      sleep 1
+    end
+
+    local_direct_statuses.find_each do |status|
+      AccountConversation.add_status(status.account, status)
+    end
+
+    notifications_about_direct_statuses.find_each do |notification|
+      AccountConversation.add_status(notification.account, notification.target_status)
+    end
+  end
```

Comment:
What about statuses being added out-of-order?

Reply:
Batch query enforces own order and it's based on id and i think it's ascending.

Reply:
Any local toot will still be added before any remote toot, regardless of their actual order, right?
This means that if the last toot of a conversation was a local toot, it won't actually show, and that removing toots from the conversation will lead to strange behaviors.
It also means that for servers running on master or the previous RC, old toots will pop up on top of existing conversations.

Reply:
You forget about the status_ids are sorted before being saved, so inserting out of order should be fine. Updates will be streamed though but only if you are connected to the direct stream and I think even then it's a minor one-off issue.

Reply:
I'm sorry, I can't find where they are sorted.

Reply:
https://github.com/tootsuite/mastodon/blob/master/app/models/account_conversation.rb#L99-L102